### PR TITLE
Cleaning the API

### PR DESCRIPTION
**Pull request by sofianehaddad**
Within the add of linear model classes from the module, it seems better to quickly fix some issues:

 - Get same method names as other classes in the lib (getCoefficients instead of getTrendCoefficients for example)
 - Move some methods from LinearModelAnalysis to LinearModelResult because these might be needed at this level
 - Fix some doc issues (openturns#1036 & openturns#1078) 

There isn't any new method

### CODE REVIEW

File: python/src/KrigingAlgorithm_doc.i.in [hunk 1]
```diff
@@ -205,10 +205,22 @@ outputSample : :class:`~openturns.Sample`
...
+  - Output dimension is :math:`\geq 2`. In that case, we get the **full** log-likelihood function :math:`\mathcal{L}(\vect{\theta}, \vect{\sigma})`.
+
+  - Output dimension is **1** and the `GeneralLinearModelAlgorithm-UseAnalyticalAmplitudeEstimate` key of :class:`~openturns.ResourceMap` is set to *True*.
+    The amplitude parameter of the covariance model :math:`\vect{\theta}` is in the active set of parameters and thus we get the **reduced**
+    log-likelihood function :math:`\mathcal{L}(\vect{\theta})`.
+
+  - Output dimension is **1** and the `GeneralLinearModelAlgorithm-UseAnalyticalAmplitudeEstimate` key of :class:`~openturns.ResourceMap` is set to *False*.
+    In that case, we get the **full** log-likelihood :math:`\mathcal{L}(\vect{\theta}, \vect{\sigma})`.
+
 The reduced log-likelihood function may be useful for some pre/postprocessing: vizuaisation of the maximizer, use of an external optimizers to maximize the reduced log-likelihood etc.
```
Comment (jschueller) [inline python/src/KrigingAlgorithm_doc.i.in]: can you fix the typo: vizuaisation

Comment (sofianehaddad) [reply python/src/KrigingAlgorithm_doc.i.in]: indeed

**Comment by regislebrun** [comment]
+1 The renamed methods were not part of OT1.12, right?

**Comment by jschueller** [comment]
nope; they came from the otlm module integration

**Comment by sofianehaddad** [comment]
no all the methods are new, that's why I am trying to fix them before the release

**Pull request merged**